tamr_unify_client/client.py

```diff
@@ -1,62 +1,76 @@
+import logging
+from typing import Optional
 from urllib.parse import urljoin
 
 import requests
-from requests import Response
+import requests.auth
+import requests.exceptions
 
 from tamr_unify_client.dataset.collection import DatasetCollection
 from tamr_unify_client.project.collection import ProjectCollection
 
-# monkey-patch Response.successful
+logger = logging.getLogger(__name__)
 
 
-def successful(self):
-    """Checks that this response did not encounter an HTTP error (i.e. status code indicates success: 2xx, 3xx).
+def successful(response: requests.Response) -> requests.Response:
+    """Checks that this response did not encounter an HTTP error.
 
-    :raises :class:`requests.exceptions.HTTPError`: If an HTTP error is encountered.
-    :return: The calling response (i.e. ``self``).
-    :rtype: :class:`requests.Response`
+    HTTP error codes match 4xx or 5xx.
+
+    Returns:
+        The response being checked.
+
+    Raises:
+        requests.exceptions.HTTPError: If an HTTP error is encountered.
     """
-    self.raise_for_status()
-    return self
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        logger.error(f"HTTP error code response body: {e.response.text}")
```

Comment:
I find the phrasing "HTTP error code response body" confusing; maybe just "HTTP Error body: {e.response.text}"

Reply:
Also, note that `requests.response.raise_for_status` can raise URLError (presumably for a Mal-formed URL and the like), and that case is not handled here (which seems fine).

Reply:
Or maybe: `"HTTP Error {e.response.status_code}. Response body: {e.response.text}`?

Reply:
I don't see how `raise_for_status` directly raises a `URLError` from looking at the [source code](https://requests.readthedocs.io/en/master/_modules/requests/models/#Response.raise_for_status).